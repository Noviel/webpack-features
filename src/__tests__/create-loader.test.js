import createLoader from '../loaders/create-loader';

describe('createLoader', () => {
  it('should throw with an undefined name', () => {
    expect(() => createLoader())
      .toThrow();
  });

  it('should throw with a non-string name', () => {
    expect(() => createLoader(420))
      .toThrow();
  });

  it('should override name with a string', () => {
    expect(createLoader('name', {}, { nameOverride: 'override' }))
      .toMatchSnapshot();
  });

  it('should override name with a function', () => {
    expect(createLoader('name', {}, { 
      nameOverride: name => `${name}-override`
    }))
      .toMatchSnapshot();
  });

  it('should return correct loader name', () => {
    expect(createLoader('strange-name-2').loader)
      .toMatchSnapshot();
  });

  it('should return correct options', () => {
    expect(createLoader('name', { a: { option: true }, b: 'option' }))
      .toMatchSnapshot();
  });
});
