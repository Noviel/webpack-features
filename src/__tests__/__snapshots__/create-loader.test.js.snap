// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createLoader should override name with a function 1`] = `
Object {
  "loader": "name-override",
  "options": Object {},
}
`;

exports[`createLoader should override name with a string 1`] = `
Object {
  "loader": "override",
  "options": Object {},
}
`;

exports[`createLoader should return correct loader name 1`] = `"strange-name-2"`;

exports[`createLoader should return correct options 1`] = `
Object {
  "loader": "name",
  "options": Object {
    "a": Object {
      "option": true,
    },
    "b": "option",
  },
}
`;
